systemd/gssproxy.service.in

```diff
@@ -5,30 +5,50 @@ After=syslog.target network.target
 Before=rpc-gssd.service
 
 [Service]
-StateDirectory=gssproxy/clients gssproxy/rcache
+ConfigurationDirectory=gssproxy
+StateDirectory=gssproxy gssproxy/clients gssproxy/rcache
 Environment=KRB5RCACHEDIR=/var/lib/gssproxy/rcache
 ExecStart=@sbindir@/gssproxy -i
 # This can be changed to notify-reload and ExecReload= can be removed once
 #   systemd 253 is common enough
 Type=notify
 ExecReload=/bin/kill -HUP $MAINPID
 
-ProtectSystem=full
-ProtectClock=true
-ProtectKernelLogs=true
-ProtectControlGroups=true
-RestrictRealtime=true
-# NoNewPrivileges: when true breaks the ability to open a socket
-#   under /var/lib/gssproxy so no NoNewPrivileges
-# PrivateTmp: can't be used as it hides ccaches stored in /tmp
-# ProtectHome: blocks access to /home which may hold ccaches
-# ProtectHostname: blocks propagation of hostname on change
-#   but in some cases, when using a keytab, we may want to see hostname
-#   changes as the server will want to respond only for the system name
-# ProtectKernelTunables: blocks ability to write to proc.
-#   on startup gssproxy needs to write in proc to let nfsd know it can
-#   use the "new" gssproxy method instead of the old rpc stuff.
-
+ProtectSystem=strict
+# Hides ccaches stored in /tmp
+PrivateTmp=no
+PrivateDevices=yes
+PrivateNetwork=yes
+PrivateIPC=yes
+# Blocks access to /home which may hold ccaches
+PrivateUsers=no
+ProtectHome=read-only
+# Blocks propagation of hostname on change but when using a keytab, we want to
+# see hostname changes as the server will want to respond only for that name
+ProtectHostname=no
+ProtectClock=yes
+# Does *not* block rw access to /proc/net/rpc/use-gss-proxy
+ProtectKernelTunables=yes
+# Blocks access to /proc/net/rpc/use-gss-proxy and executable name matching
+ProtectProc=default
+ProtectKernelModules=yes
+ProtectKernelLogs=yes
+ProtectControlGroups=yes
+RestrictAddressFamilies=AF_UNIX AF_LOCAL
+RestrictNamespaces=yes
```

Comment:
Not sure this is correct, when there are multiple name spaces created by containers we want the gssproxy to be working within the correct namespace too if they mount the socket.
I haven't fully read what RestrictNamespaces do, so I may just be grasping at straws here.

Reply:
I'm fairly certain that the patch will break some systems given all the hardening options. I'll have a look at the containers/namespaces scenario

Reply:
Running nfs server correctly for different namespaces is one of the goals of gssproxy.
If this needs service file created for specific namespaces we can document it, and leave the main service file more restricted as well.
But if we can make the main service file compatible with multiple nfs servers/clients in different namespaces we should try.

Reply:
> Running nfs server correctly for different namespaces is one of the goals of gssproxy. If this needs service file created for specific namespaces we can document it, and leave the main service file more restricted as well. But if we can make the main service file compatible with multiple nfs servers/clients in different namespaces we should try.

Does `gssproxy` muck around with namespaces or it is just launched *in* a given namespace? After some quick grepping I can't find code that would actually change namespaces (`unshare`, `setns`, `clone`)....? If so, the `RestrictNamespaces=` should be ok...since it's about restricting `ns` *changes*, not `ns` support per se...